EKF/control.cpp

```diff
@@ -492,16 +493,19 @@ void Ekf::controlGpsFusion()
 
 	} else {
 		// handle the case where we do not have GPS and have not been using it for an extended period, but are still relying on it
-		if ((_time_last_imu - _time_last_gps > 10e6) && (_time_last_imu - _time_last_airspeed > 1e6) && (_time_last_imu - _time_last_optflow > 1e6) && _control_status.flags.gps) {
+		if ((_time_last_imu - _time_last_gps > 10e6) && _control_status.flags.gps) {
 			// if we don't have a source of aiding to constrain attitude drift,
 			// then we need to switch to the non-aiding mode, zero the velocity states
 			// and set the synthetic GPS position to the current estimate
 			_control_status.flags.gps = false;
-			_last_known_posNE(0) = _state.pos(0);
-			_last_known_posNE(1) = _state.pos(1);
-			_state.vel.setZero();
-			ECL_WARN("EKF measurement timeout - stopping navigation");
+			ECL_INFO("EKF Stopping GPS Use");
 
+			if (!_control_status.flags.opt_flow && !_control_status.flags.ev_pos) {
```

Comment:
The concern I have with this is that it removes the ability of a FW aircraft using airspeed fusion to continue to dead-reckon and provide an attitude estimate.